packages/contracts/src/ERC1271SignatureValidator.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract ERC1271SignatureValidator {
+    /// Mapping to store if a hash has been signed.
+    mapping(bytes32 => bool) public isHashSigned;
```

Comment:
1. Where is this updated?
2. What do you think about verifying proof etc directly in `isValidSignature`? And state updates can be handled in higher level logic such as how contract signatures work on Safes https://github.com/safe-global/safe-smart-account/blob/1c8b24a0a438e8c2cd089a9d830d1688a47a28d5/contracts/Safe.sol#L140-L142

This way the `EmailSigner` can be used to verify "signatures" in a single step rather than approve + verify